Merge in Select mpa-sil Tweaks

nikheizen opened the pull request:
I think these are all reasonable and minor changes made in the mpa-sil fork.

**Interface:**
- Glyph changes
- Weights of floor items
- Prompt to step beside immobile monsters
- Force uppercase Y/N on vi keys
- Running stops in doorways

**Convenience:**
- Starting with the sword equipped and always the same weight
- IDing glowing weapons, booze, and healing items
- Not wasting turns (for empty staves, when picking up with a full inventory)
- Allowing melting mithril in exhausted forges
- Regeneration not consuming hunger at full HP

**Dumps/Inscriptions:**
- Killed/Seen counts
- Artifact and skill areas
- Consumable inscriptions
- Automatic dumps

Of these, I think that the only questionable one may be the "prompt on moving next to immobiles" since statdrain molds are no longer a thing, and the only significant immobile thing to move beside now is a yellow & during a fight.

sil-quirk commented:
Quite a bit to unpack here.

+ Glyph changes
- I don't see any changes related to these?

+ Weights of floor items
- I'm torn on this. I don't believe you should necessarily be able to determine weight simply from looking at an item. Willing to discuss. Maybe.

+ Prompt to step beside immobile monsters
- No, I took this for a brief run and I think this is annoying. The only immobile monsters are the thorns on early levels, and I clear a couple for the XP and otherwise let them be. I've never run into them by accident save when confused. Since this would annoy me personally it's not going in. No.

+ Force uppercase Y/N on vi keys
- This seems broadly a good idea. Yes.

+ Running stops in doorways
- Bug hadn't really penetrated my consciousness before, but it seems a good fix. Yes.

+ Starting with the sword equipped and always the same weight
- I plan to mess with starting equipment at some point, but this isn't something I care about particularly right now one way or the other. Yes, if you want.

+ IDing glowing weapons, booze, and healing items
- Glowing weapon ID looks bugged to me; it doesn't check whether the identification could be ambiguous (i.e. if you're surrounded by wolves and there's an orc in view, it'll correctly identify as Gondolin). This could be made to work, but it should be a dedicated commit which handles this more delicately. No for now.
- AutoID of Miruvor and Orcish Liquor - no very strong opinion here. Line between character and player knowledge blurry in any case. I note the fix uses spaces in an area with tabs? But I should probably clang-format the whole damn mess. Yes, if you want.
- ID healing at full. Only if we add a message indicating some kind of fuzzy good feeling. "You feel full" followed by "You have no more X of Healing" doesn't scratch my UI itch. Yes, with caveat.

+ Not wasting turns (for empty staves, when picking up with a full inventory)
- Seems good. Yes.
 
+ Allowing melting mithril in exhausted forges
- Also fine. Yes.

+ Regeneration not consuming hunger at full HP
- I'm a little torn; it depends how regeneration works. If it's simply a faster metabolism that causes faster healing, hunger should be on all the time. If healing pulls more energy... Not sure. However, regeneration is currently somewhat niche, so boosting it a little isn't really problematic. Yes, if you want.

+ Killed/Seen counts
- Yes, absolutely fine.

+ Artifact and skill areas
- Yes.

+ Consumable inscriptions
- Depth? Not sure I like this or see the need of it. Feels metagamey. Open to convincing. Maybe.

+ Automatic dumps
- Yes.

nikheizen commented:
Alright. I went in and squashed a few commits together. History has been slightly rewritten, but looks better as a result (still a bit messy).

**Changed/Removed commits:**

-Removed the prompt-for-immobile check and the Y/N uppercase prompt commits.
-Removed glowing weapon ID changes.
-Added a message for drinking/eating Healing. (went with simple, could change, of course)

**Added commits:**

-Stair recolouring/door glyphs.
-ID Belegost/=Regen on equip.
-Deprecate destroy command, make picking up the crown prise sils.
-Silcutting message/endgame message typo fixes.
-Remove the debug password

What remains unported is stuff you have said "no" to and most of the debug mode fixes from mpa-sil which I have not gone over yet.

Weight of floor items: so that you don't have to shuffle through items repeatedly after killing a pack, stepping over every one multiple times to check the weight.

Regeneration: the vanilla implementation provokes the player to remove it while they are at full HP/wandering so as not to get hungry faster, then put it back on when about to fight. This is simply annoying and the boost is pretty minor as food is mostly a non-issue.

Depth inscription on consumables: It is definitely metagamey, but I like not having to write down/inscribe the depth I found stuff at myself, so I can figure "these are probably antidote", "this one is probably a statpot", etc.

I was however talking with absolutego recently who mentioned the idea of untying consumable item distribution from depth entirely, which would make this change completely unnecessary. I like the way DCSS does it, which is just a table of item weights for each item in a class. I don't know how hard this would be to implement & probably not the best place to discuss it.

sil-quirk commented:
I don't see the healing ID commit now? You've convinced me on regeneration, good point about incentivising player gear swaps. I think I would like the depth commit pulled, and the weight commit -weight is only really a concern on primary weapon and for Heavy Armour Use, and I'm not sure that's enough for me to consent to an uglier interface. I'm fine with Belegost for now though I note that the reason given is that it is unambiguous (not quite true as there is a mattock artefact but close enough) and this will need changed if additional egos for mattocks are created.

sil-quirk commented:
Okay, since I've got a couple of fairly substantial bugs to patch in the next release and am a little in a hurry, I pulled in most of the stuff we were both happy with (did a version of the Healing auto-id although the commit seemed to be missing). Do you mind if I decline this pull request, then you can review the changes I've adopted and we can talk about some of the other things in here in a new request?

nikheizen commented:
Sounds good.
